web_generator/lib/src/banned_names.dart

```diff
@@ -15,8 +15,11 @@ const bannedNames = <String>{
 
 /// Given a [jsName], returns the allowed Dart equivalent accounting for invalid
 /// characters and reserved keywords.
-String dartRename(String jsName) {
+String dartRename(String jsName, [bool toSingleForm = false]) {
   var dartName = jsName.replaceAll('-', '_');
+  if (toSingleForm && dartName.length > 2 && dartName.endsWith('s')) {
```

Comment:
I like this in concept, but I'm not sure there's anything guaranteeing that a variadic arguments name can't end in an `s` unless it's plural e.g. `class`. It looks like this PR doesn't come across that issue, and if it does, changing names isn't a breaking change, so this is fine with me.

Reply:
Another idea is to create a dict and fill it with new argument names, they are mostly the same.

Reply:
That's okay - I think the current code is fine with me, resolving.